src/main/java/gregtech/api/metatileentity/multiblock/MultiblockWithDisplayBase.java

```diff
@@ -4,21 +4,261 @@
 import gregtech.api.gui.ModularUI;
 import gregtech.api.gui.Widget.ClickData;
 import gregtech.api.gui.widgets.AdvancedTextWidget;
+import gregtech.api.multiblock.IMaintenance;
+import gregtech.api.multiblock.PatternMatchContext;
+import gregtech.api.unification.OreDictUnifier;
+import gregtech.api.unification.material.Materials;
+import gregtech.api.unification.ore.OrePrefix;
+import gregtech.api.util.XSTR;
+import gregtech.common.ConfigHolder;
+import gregtech.common.metatileentities.electric.multiblockpart.MetaTileEntityMaintenanceHatch;
+import gregtech.common.metatileentities.electric.multiblockpart.MetaTileEntityMufflerHatch;
```

Comment:
Again here should use the interfaces to keep as much of common out of API (config is often unavoidable)